src/scripts/infuse/dimInfuse.controller.js

```diff
@@ -5,9 +5,23 @@ angular.module('dimApp')
   .controller('dimInfuseCtrl', dimInfuseCtrl);
 
 
-function dimInfuseCtrl($scope, dimStoreService, dimItemService, ngDialog, dimLoadoutService, toaster, $q, $translate) {
+function dimInfuseCtrl($scope, dimStoreService, dimDefinitions, ngDialog, dimLoadoutService, toaster, $q, $translate) {
   var vm = this;
 
+  vm.i18n = {};
+  vm.icon = {};
+  dimDefinitions.getDefinitions().then((defs) => {
+    vm.i18n[452597397] = defs.InventoryItem.get(452597397).itemName;
+    vm.i18n[2534352370] = defs.InventoryItem.get(2534352370).itemName;
+    vm.i18n[3159615086] = defs.InventoryItem.get(3159615086).itemName;
+    vm.i18n[937555249] = defs.InventoryItem.get(937555249).itemName;
+    vm.i18n[1898539128] = defs.InventoryItem.get(1898539128).itemName;
+    vm.i18n[1542293174] = defs.InventoryItem.get(1542293174).itemName;
+    vm.icon[937555249] = defs.InventoryItem.get(937555249).icon;
+    vm.icon[1898539128] = defs.InventoryItem.get(1898539128).icon;
+    vm.icon[1542293174] = defs.InventoryItem.get(1542293174).icon;
```

Comment:
Rather than assigning these properties individually, you could just save the item definition itself indexed by hash, and access the sub-properties directly in the view.